### ISSUE TITLE
Spend confirmation is more difficult then it needs to be. Please allow fingerprint confirmations.

### ISSUE THREAD

**Issue by PC-Admin** [m2049r/xmrwallet#571]
Been trying Monerujo in the wild.

You can unlock the wallet with your fingerprint, great. But when it comes time to spend your left awkwardly reaching into your wallet and typing out a long password.

Please allow spend confirmations with the fingerprint sensor. Tap-to-pay systems are like at least 100x faster than this app right now.

Cheers

**Comment by m2049r** [comment]
the current idea is to let you into your wallet easily (through fingerprint), but when you want to spend, you need to really id yourself - through the password.

spending is not possible in bankings apps - one time tan/password is always required. afaict all (?) wallet apps need the password as well for spending.

don't know how relevant this may be, but also consider that most jurisdictions can force you to use your fingerprint. but not to reveal a password.

you may want to think about having a wallet with a "low" everyday balance without a password... (or a simple password)

**Comment by PC-Admin** [comment]
Hi m2049r,

> the current idea is to let you into your wallet easily (through fingerprint), but when you want to spend, you need to really id yourself - through the password.

A fingerprint is stronger then the passwords most people use, it also acts as an identifier imo.

> spending is not possible in bankings apps - one time tan/password is always required. afaict all (?) wallet apps need the password as well for spending.

I'm not saying you shouldn't still set a strong password to encrypt the wallet file, but it would be a major upgrade if you had the **option** of spending with your fingerprint instead. Monero would suddenly have one of the most usable mobile wallets in the crypto space! :)

> don't know how relevant this may be, but also consider that most jurisdictions can force you to use your fingerprint. but not to reveal a password.

Authorities in most countries can compel you (with jail time) to give both your fingerprint and any passwords you may have set. (Even if you claim not to know them.) Also since you can already unlock the wallet and see the balance and previous transactions with your fingerprint there isn't much more data they can get by asking for your wallet password. I would not carry XMR that i was not willing to declare anyway, certainly not in a mobile wallet that can be easily seen!

> you may want to think about having a wallet with a "low" everyday balance without a password... (or a simple password)

That's not ideal since a simple password is both weaker then a fingerprint and still considerably more difficult to use. :(

I would like to reinstate my interest in having an 'optional fingerprint spend confirmation feature'.

Thank you, (tip sent)

**Comment by jonathancross** [comment]
> most jurisdictions can force you to use your fingerprint. but not to reveal a password.

I agree with @PC-Admin that the password is not adding much in this attack scenario (law enforcement officers who already can see balance, transaction history, etc).  If they are not corrupt, they will not try to steal your Monero (making the password useless).  If they **are** corrupt, then password is no stronger than your fingerprint (laws about what can or cannot be demanded don't matter in this situation).

That being said, what is the point of having a strong password that is never required?

Personally, I have also struggled with this issue having setup a unique, complex password for Monerujo, then being surprised when asked for it at a point-of-sale. However, I now see that as my mistake rather than the apps.  The more I think through the various scenarios, the more that the current model makes sense to me.

Maybe there is something that can be added during the wallet fingerprint setup process to suggest that the user can change to a weak (easy to remember) password or PIN if lower security is acceptable.

>> you may want to think about having a wallet with a "low" everyday balance without a password... (or a simple password)

> That's not ideal since a simple password is both weaker then a fingerprint and still considerably more difficult to use. :(

This suggestion from @m2049r seems like a good workaround, but not one that was obvious to me at first.  @PC-Admin: Why not set the password to your phone's PIN number (very easy) and _also_ use the fingerprint?  Seems sufficient for low-balance spending wallet.

**Comment by m2049r** [comment]
may an "allow fingerprint spending" option with a user-defined max xmr limit would be helpful?

then password would be required to see the secrets / change password options and fingerprint could be used to open the wallet and spend "small" amounts.

EDIT: and we would also need some time-contraint on these small amounts to prevent the person who stole your finger to deplete the wallet with a bunch of small payments.

**Comment by PC-Admin** [comment]
>  may an "allow fingerprint spending" option with a user-defined max xmr limit would be helpful?

That would be awesome, you could define the limit as a percentage so 0% = off and 100% means you are okay fingerprinting away the entire balance. Or as just as a plain value.

Thank you for considering this.
